changelog.d/kafka-source-parse_message.feat.md

```diff
@@ -0,0 +1 @@
+The constructor of FrameRead will allocate an initial memory of 8192, In cases where Kafka messages are smaller than this value, additional memory will be used. Initializing it with msg.payload_len() would be more appropriate.
```

Comment:
```suggestion
The `kafka` source will now consume less memory by not over allocating buffers while reading.

authors: biuboombiuboom
```

These changelog entries are intended for Vector users rather than developers so we usually don't include implementation details.